### ISSUE TITLE
[batch] Help users deal with duplicate outputs

### ISSUE THREAD

**Issue by danking** [hail-is/hail#13502]
### What happened?

Since we guarantee a job will run at least once, there are two issues that can happen:

1. A user can write a pipeline in which two jobs race to write the same file, e.g.
   ```
   j = b.new_job()
   j.command('echo hello > {j.out}')
   j.write_output(j.out, "gs://bucket/final-output")
   ```
2. Or, a clever user can avoid this race with some randomness:
   ```
   j = b.new_job()
   j.command('echo hello gsutil cp - gs://bucket/final-output-$RANDOM')
   ```

The former is a really common pattern and a bit of a footgun! The latter is rare (I don't know anyone who does it) and hard to work with: how would you know the output file of the *successful* attempt?

Hail should provide some mechanism for a user to get the list of successful attempts and their outputs. One simple option is to include some kind of seeded randomness which the user can access and to return either the seed or all the draws of the successful attempt for each job in `/jobs` or for the one job in `/job/{job_id}`.

For example, consider:

```
j = b.new_job()
j.command('echo hello gsutil cp - gs://bucket/final-output-$(/hail-random-str)')
```

Where `/hail-random-str` is a binary we mount into the container that randomly generates numbers seeded by `(batch id, job id, attempt id)`. Hail should use the same randomness to ensure that `write_output` is reliable.

### Version

0.2.120

**Comment by patrick-schultz** [comment]
If I'm understanding the example at the end correctly, it sounds like you really only need a hash of `(batch id, job id, attempt id)` (though hash functions and PRNGs are very closely related).

**Comment by danking** [comment]
Yeah that's sufficient for most cases, but I was anticipating the need to write more than one file with distinct names. I guess they can always do RAND-1, RAND-2, ... though.

**Comment by patrick-schultz** [comment]
The hashes are appended to the end of the filename. Shouldn't that be enough to ensure distinct names?

**Comment by danking** [comment]
Yeah I think you’re right
